Backend/app/dashboard/Lib/src/Tools/Auth/Dashboard.hs

```diff
@@ -62,16 +66,24 @@ instance VerificationMethodWithPayload VerifyDashboard where
   type VerificationPayloadType VerifyDashboard = DRole.DashboardAccessType
 
 verifyDashboardAction ::
-  (Common.AuthFlow m r, Redis.HedisFlow m r) =>
+  (Common.AuthFlow m r, Redis.HedisFlow m r, HasFlowEnv m r '["dashboardTokenExpiry" ::: Seconds]) =>
   VerificationActionWithPayload VerifyDashboard m
 verifyDashboardAction = VerificationActionWithPayload verifyDashboard
 
 verifyDashboard ::
-  (Common.AuthFlow m r, Redis.HedisFlow m r) =>
+  (Common.AuthFlow m r, Redis.HedisFlow m r, HasFlowEnv m r '["dashboardTokenExpiry" ::: Seconds]) =>
   DRole.DashboardAccessType ->
   RegToken ->
   m TokenInfo
 verifyDashboard requiredAccessType token = do
+  registration <- RT.findByToken token >>= fromMaybeM (InvalidToken token)
```

Comment:
This check should be inside `verifyPerson` function where we're caching the token in redis.